src/java/com/synacor/zimbra/OAuthDataImport.java

```diff
@@ -0,0 +1,35 @@
+package com.synacor.zimbra;
+import com.synacor.zimbra.ys.contacts.YahooContactsImport;
+import com.zimbra.cs.account.DataSource;
+
+import java.util.List;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.account.DataSource.DataImport;
+/**
+ * @author Greg Solovyev
+ * DataImport class that picks source-specific implementation based on oauth2 URL. 
+ * This class will only be aware of implementations packaged together with this class.
+ */
+public class OAuthDataImport implements DataImport {
+    private DataImport implementation;
+    public OAuthDataImport(DataSource ds) {
+        String sourceURL = ds.getOauthRefreshTokenUrl();
+        if(sourceURL.indexOf("api.login.yahoo.com/oauth2/get_token") > -1) {
```

Comment:
Let's find another way to determine which service we need. Something more generic so it's easier to reuse this stuff for syncs beyond contacts (e.g. calendar, profile, photos, ...).

Also let's talk about minimizing storage of the refreshToken url since it doesn't change by user.